crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -186,11 +188,38 @@ impl C8YHttpProxyActor {
         &mut self,
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
-        let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let url_get_id: String = self.end_point.get_url_for_get_id(device_id);
+        let req_build_closure = |token: String| {
+            let request_internal_id = HttpRequestBuilder::get(&url_get_id);
+            request_internal_id.bearer_auth(token)
```

Comment:
As adding a token is done for all and every request and as this can be done blindly, it's better to move this inside the `execute` method.

Reply:
fixed